e3sm_diags/derivations/derivations.py

```diff
@@ -0,0 +1,1484 @@
+"""This module stores definitions for derived variables.
+
+`DERIVED_VARIABLES` is a dictionary that stores definitions for derived
+variables. The driver uses the Dataset class to search for available variable
+keys and attempts to map them to a formula function to calculate a derived
+variable.
+
+For example to derive 'PRECT':
+  1. In `DERIVED_VARIABLE` there is an entry for 'PRECT'.
+  2. The netCDF file does not have a 'PRECT' variable, but has the 'PRECC'
+      and 'PRECT' variables.
+  3. 'PRECC' and 'PRECL' are used to derive `PRECT` by passing the 
+  	  data for these variables to the formula function 'prect()'.
+"""
+from collections import OrderedDict
+from typing import Callable, Dict, Tuple
+
+from e3sm_diags.derivations.formulas import (
+    albedo,
+    albedo_srf,
+    albedoc,
+    fldsc,
+    flus,
+    fp_uptake,
+    fsus,
+    lwcf,
+    lwcfsrf,
+    molec_convert_units,
+    netcf2,
+    netcf2srf,
+    netcf4,
+    netcf4srf,
+    netflux4,
+    netflux6,
+    netlw,
+    netsw,
+    pminuse_convert_units,
+    precst,
+    prect,
+    qflx_convert_to_lhflx,
+    qflx_convert_to_lhflx_approxi,
+    qflxconvert_units,
+    restoa,
+    restom,
+    rst,
+    rstcs,
+    swcf,
+    swcfsrf,
+    tauxy,
+    tref_range,
+    w_convert_q,
+)
+from e3sm_diags.derivations.utils import (
+    _apply_land_sea_mask,
+    aplusb,
+    convert_units,
+    cosp_bin_sum,
+    cosp_histogram_standardize,
+    rename,
+)
+
+# A type annotation ordered dictionary that maps a tuple of source variable(s)
+# to a derivation function.
+DerivedVariableMap = OrderedDict[Tuple[str, ...], Callable]
+
+# A type annotation for a dictionary mapping the key of a derived variable
+# to an ordered dictionary that maps a tuple of source variable(s) to a
+# derivation function.
+DerivedVariablesMap = Dict[str, DerivedVariableMap]
+
+
+DERIVED_VARIABLES: DerivedVariablesMap = {
+    "PRECT": OrderedDict(
+        [
+            (
+                ("PRECT",),
+                lambda pr: convert_units(rename(pr), target_units="mm/day"),
+            ),
+            (("pr",), lambda pr: qflxconvert_units(rename(pr))),
+            (("PRECC", "PRECL"), lambda precc, precl: prect(precc, precl)),
+        ]
+    ),
+    "PRECST": OrderedDict(
+        [
+            (("prsn",), lambda prsn: qflxconvert_units(rename(prsn))),
+            (
+                ("PRECSC", "PRECSL"),
+                lambda precsc, precsl: precst(precsc, precsl),
+            ),
+        ]
+    ),
+    # Sea Surface Temperature: Degrees C
+    # Temperature of the water, not the air. Ignore land.
+    "SST": OrderedDict(
+        [
+            # lambda sst: convert_units(rename(sst),target_units="degC")),
```

Comment:
Why is this comment here?

Reply:
Not sure, this dictionary was pulled from `acme.py` into `derivations.py` without being refactored yet (should be handled in #716).

Reply:
"SST" is Sea Surface Temperature with land masked. I guess that's why the comment is there..

Reply:
OH, I think this refers to `# lambda sst ...` this can be cleaned up in later refactor.

Reply:
Thanks for the clarification Jill